Flickr/Flickr/ImageCollectionViewController.swift

```diff
@@ -82,20 +94,46 @@ class ImageCollectionViewController: UIViewController,  UICollectionViewDelegate
         setDelegates_DataSources()
         setAlphasDefault()
         setConstraintMode()
-        setBehaviorTextField()
+        setBehaviorTextBar()
+        loadFormUserDefaults()
         setXibCellForRecentTableViewCell()
+        searchTextField.clearButtonMode = .always
+        customPullToUpdate()
+    }
+    
+    func loadFormUserDefaults() {
+        guard let defaults = UserDefaults.standard.array(forKey: "historySearch") else {
+            return
+        }
+        if !defaults.isEmpty {
+            searchHistoryList = defaults as! [String]
+        }
+      
     }
     
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        rebuildTableSize()
-        performFlickrPopular()
-        
+        if searchImageData.isEmpty && popularImageData.isEmpty {
+            rebuildTableSize()
+            performFlickrPopular()
+            return
+        }
     }
     
     override func viewDidLayoutSubviews(){
         rebuildTableSize()
-        searchHistoryView.reloadData()
+        searchHistoryTableView.reloadData()
+    }
+    
+    func customPullToUpdate() {
+        let refreshViewSearch = UIView(frame: CGRect(x: 0, y: 62, width: 0, height: 0))
+        let refreshViewPopular = UIView(frame: CGRect(x: 0, y: 62, width: 0, height: 0))
+        searchCollectionView.addSubview(refreshViewSearch)
+        popularCollectionView.addSubview(refreshViewPopular)
+        refreshControlForSearch.addTarget(self, action: #selector(refreshCurrentCollectionViewByPullToUpdate), for: .valueChanged)
+        refreshControlForPopular.addTarget(self, action: #selector(refreshCurrentCollectionViewByPullToUpdate), for: .valueChanged)
```

Comment:
make separate refreshCurrentCollectionViewByPullToUpdate method for 2 collections